src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/OAuth2LoginBackingBean.java

```diff
@@ -79,60 +143,36 @@ public void exchangeCodeForToken() throws IOException {
                 }
                 error = new OAuth2Exception(-1, sb.toString(), "Remote system did not return an authorization code.");
                 logger.log(Level.INFO, "OAuth2Exception getting code parameter. HTTP return code: {0}. Message: {1} Message body: {2}", new Object[]{error.getHttpReturnCode(), error.getLocalizedMessage(), error.getMessageBody()});
-                return;
-            }
-        }
-
-        final String state = req.getParameter("state");
-
-        try {
-            AbstractOAuth2AuthenticationProvider idp = parseState(state);
-            if (idp == null) {
-                throw new OAuth2Exception(-1, "", "Invalid 'state' parameter.");
-            }
-            oauthUser = idp.getUserRecord(code, state, getCallbackUrl());
-            UserRecordIdentifier idtf = oauthUser.getUserRecordIdentifier();
-            AuthenticatedUser dvUser = authenticationSvc.lookupUser(idtf);
-            
-            if (dvUser == null) {
-                // need to create the user
-                newAccountPage.setNewUser(oauthUser);
-                FacesContext.getCurrentInstance().getExternalContext().redirect("/oauth2/firstLogin.xhtml");
-
-            } else {
-                // login the user and redirect to HOME of intended page (if any).
-                session.setUser(dvUser);
-                session.configureSessionTimeout();
-                
-                final OAuth2TokenData tokenData = oauthUser.getTokenData();
-                tokenData.setUser(dvUser);
-                tokenData.setOauthProviderId(idp.getId());
-                oauth2Tokens.store(tokenData);
-                String destination = redirectPage.orElse("/");
-                HttpServletResponse response = (HttpServletResponse) FacesContext.getCurrentInstance().getExternalContext().getResponse();
-                String prettyUrl = response.encodeRedirectURL(destination);
-                FacesContext.getCurrentInstance().getExternalContext().redirect(prettyUrl);
+                return Optional.empty();
+            } catch (IOException e) {
+                error = new OAuth2Exception(-1, "", "Could not parse OAuth2 code due to IO error.");
```

Comment:
More English.

Reply:
This exception is never thrown, but logged as localized message. Nothing TODO :smile: